maas/plugins/process_check.py

```diff
@@ -0,0 +1,121 @@
+    return set([comm, process_exe, os.path.basename(process_exe)])
+
+
+def get_processes(procs_path):
+    process_names = set()
+    try:
+        with open(procs_path, mode='r') as f:
+            for pid in f.read().splitlines():
+                process_names.update(get_process_name(pid))
+    except OSError as ex:
+        # Could not read the PID list file, probably due to a permission
+        # error. This is different from an IOError, so we handle the two
+        # types differently.
+        status_err('Could not read PID list file. {}', exception=ex)
+    except IOError as ex:
+        # The path we are checking for a PID list does not exist.
+        # This is probably caused by an invalid container name.
+        status_err('No such container exists. {}', exception=ex)
+
+    # Return a set() of potential process names resolved from their PID
+    return process_names
+
+
+def check_process_running(process_names, container_name=None):
+    """Check to see if processes are running.
+
+       Check if each of the processes in process_names are in a list
+       of running processes in the specified container name, or on
+       this host.
+    """
+
+    if not process_names:
+        # The caller has not provided a value for process_names, which gives us
+        # nothing to do. Return an error for the check.
+        status_err('No process names provided')
+
+    procs = []
+    if container_name is None:
+        # Checking for processes on a host, not inside a container
+        procs_path = '/sys/fs/cgroup/cpu/cgroup.procs'
+        procs = get_processes(procs_path)
+    else:
+        # Checking for processes in a container, not the parent host
+        procs_path = os.path.join('/sys/fs/cgroup/cpu/lxc', container_name,
+                                  'cgroup.procs')
+        procs = get_processes(procs_path)
+
+    if not procs:
+        # Unable to get a list of process names for the container or host.
+        status_err('Could not get a list of running processes')
+
+    # Since we've fetched a process list, report status_ok.
+    status_ok()
+
+    # Report the presence of each process from the command line in the
+    # running process list for the host or specified container.
+    for process_name in process_names:
+        metric_bool('%s_process_status' % process_name,
+                    process_name in procs)
+
+
+def main(args):
+    if not args.processes:
+        # The command line does not have any process names specified
+        status_err('No executable names supplied')
+
+    check_process_running(container_name=args.container,
+                          process_names=args.processes)
+
+
+if __name__ == "__main__":
+    with print_output():
+        parser = argparse.ArgumentParser(
+            description='Check a host or container for running processes')
+        parser.add_argument('-c', '--container', action='store')
+        parser.add_argument('processes', nargs=argparse.REMAINDER)
```

Comment:
The way it is now, argparse will gather all remaining arguments on the command-line. I added a print statement of ``args.processes`` and pasted the output below:
```
Process names: ['filebeat', '-c', 'cantu-mitaka2_cinder_api_container-b00686ee']
```
This will cause the plugin to produce three metrics; one for each of the entries above.

```
./run_plugin_in_venv.sh process_check.py filebeat -c cantu-mitaka2_cinder_api_container-b00686ee
Process names: ['filebeat', '-c', 'cantu-mitaka2_cinder_api_container-b00686ee']
status okay
metric filebeat_process_status uint32 1
metric -c_process_status uint32 0
metric cantu-mitaka2_cinder_api_container-b00686ee_process_status uint32 0
```